nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/ui/web/download/DownloadServlet.java

```diff
@@ -88,78 +79,67 @@ protected void handleDownload(HttpServletRequest req, HttpServletResponse resp)
             resp.sendError(HttpServletResponse.SC_NOT_FOUND, "Invalid URL syntax");
             return;
         }
-        String localURI = requestURI.substring(context.length());
-        // find what to do
-        int slash = localURI.indexOf('/');
-        if (slash < 0) {
+        String uri = requestURI.substring(context.length());
+        Pair<String, String> pair = downloadService.getDownloadPathAndActionFromUri(uri);
+        if (pair == null) {
             resp.sendError(HttpServletResponse.SC_NOT_FOUND, "Invalid URL syntax");
             return;
         }
-        String what = localURI.substring(0, slash);
-        String path = localURI.substring(slash + 1);
-        switch (what) {
-        case NXFILE:
-        case NXBIGFILE:
-            handleDownload(req, resp, path, false);
-            break;
-        case NXDOWNLOADINFO:
-            // used by nxdropout.js
+        String path = pair.getLeft();
+        String action = pair.getRight();
+        switch (action) {
+        case DownloadService.INFO:
             handleDownload(req, resp, path, true);
             break;
-        case NXBIGZIPFILE:
-        case NXBIGBLOB:
-            Framework.getService(DownloadService.class).downloadBlob(req, resp, path, "download");
+        case DownloadService.EVENT_NAME_FROM_DOC:
+            handleDownload(req, resp, path, false);
+            break;
+        case DownloadService.EVENT_NAME:
+            downloadService.downloadBlob(req, resp, path, "download");
             break;
         default:
             resp.sendError(HttpServletResponse.SC_NOT_FOUND, "Invalid URL syntax");
         }
     }
 
-    protected void handleDownload(HttpServletRequest req, HttpServletResponse resp, String urlPath, boolean info)
+    protected void handleDownload(HttpServletRequest req, HttpServletResponse resp, String path, boolean info)
```

Comment:
Can't this whole method be moved to the DownloadService as well? I think it would simplify things. In particular a number of shared methods that you added wouldn't need to be part of the public API interface (you can keep them `public` though if it helps unit tests).